src/controller.rs

```diff
@@ -153,6 +159,18 @@ impl Controller for FoxBox {
         format!("ws://{}:{}/services/{}/", self.hostname, self.ws_port, service_id)
     }
 
+    fn get_config(&self, namespace: String, property: String, default: String)
+        -> String {
```

Comment:
I'd prefer keeping the return type be Option<String>. That's more Rust-y and let callers fallback as they like.